app/src/lib/dispatcher/app-store.ts

```diff
@@ -1130,9 +1131,7 @@ export class AppStore {
   }
 
   private async guessGitHubRepository(repository: Repository): Promise<GitHubRepository | null> {
-    const gitStore = this.getGitStore(repository)
-    const remote = gitStore.remote
-
+    const remote = await getDefaultRemote(repository)
```

Comment:
This change was made because in testing I found that there was sometimes a race condition where the `gitStore` hadn't finished loading yet and therefore the remote was null. We could fix this by waiting on the store to become ready but unless I'm mistaken we should, if for no other reason than consistency, try to guess the remote based on the default remote and not based on whichever branch is open at the time. If anyone has a reason why we shouldn't I'm all ears